### DISCUSSION TITLE
Refetching queries or updating subscriptions when their variables change without framework bindings

### DISCUSSION THREAD

**Discussion by tmaxmax** [urql-graphql/urql#1643] [Q&A] [answered]
Hello there! I was curious if you have some tips on how to react to variable changes when using the `@urql/core` package without any bindings. In other words, how can I refetch queries and update subscriptions with the core package?

**Comment by JoviDeCroock** [comment, marked as answer]
You can look at our framework bindings for inspiration around this, the gist of it often comes down to the same thing, you'll need some way to observe the variables object. When this callback/.... triggers because it has observed variable changes you unsubscribe from the previous subscription by calling the `unsubscribe` return value and initiate your new subscription.

In React-land the simplest equivalent would be something like:

```js
import { createRequest } from '@urql/core';
import { pipe, subscribe } from 'wonka';

const useQuery = ({ query, variables }) => {
  const [data, setData] = useState();
  const client = useClient();
  const prevRequest = useRef();

  // This ensures we aren't too eager in restarting our streams
  const request = useMemo(() => {
      const request = createRequest<Data, Variables>(query, variables);
      if (prevRequest.current !== undefined && prevRequest.current.key === request.key) {
        return prevRequest.current;
      } else {
        prevRequest.current = request;
        return request;
      }
    }, [query, variables]);
  }
  
  useEffect(() => {
    const { unsubscribe } = pipe(
      client.executeQuery(request, {}),
      subscribe(result => setResult(result)),
     );
     return unsubscribe;
  }, [request.key]);

 return result;
```

This is just a simplified example and misses things like `fetching` tracking and passing in context;
